### PR TITLE
[SNOW-843760] Update pom.xml corresponding to Wiz vulnerability scan

### PR DESCRIPTION
**Pull request by sfc-gh-rcheng**
Changes, full scan in slack (ping revi for link)

**net.minidev:json-smart**: 2.4.7 -> 2.4.9
com.nimbusds.nimbus-jose-jwt references json-smart
note: can't find reference to this anymore, even using lukas's process

**com.google.protobuf:protobuf-java**: 3.7.1 -> 3.16.3 and **com.google.guava:guava**: 30.1.1-jre -> 32.0.0
org.apache.hadoop/hadoop-common (exclude io.dropwizard.metrics.metrics-core to converge dependencies)
org.apache.hadoop.thirdparty/hadoop-shaded-guava-1.1.1: 3.71
org.apache.hadoop.thirdparty/hadoop-shaded-protobuf_3_7-1.1.1.jar: 3.7.1

**com.fasterxml.jackson.core:jackson-databind**
2.13.2.2 -> 2.13.4.2
org.apache.parquet.parquet-hadoop references jackson-databind
org.apache.commons/commons-configuration2: 2.13.3
org.apache.hadoop/hadoop-common
org.apache.parquet/parquet-hadoop
parquet-jackson-1.13.1: 2.13.4.2
org.apache.parquet/parquet-jackson

### CODE REVIEW

**Review by sfc-gh-tjones — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -354,6 +358,7 @@
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
+      <version>2.14.0</version>
```
[the remark this answers is not in the document]
Comment (sfc-gh-tzhang) [reply pom.xml:361]: +1

**Review by sfc-gh-tzhang — APPROVED** [review]
Thanks Revi! Do you think we should provide a private jar to CITI or Informatica and ask them to run it with Blackduck in case there is still some differences? My concern is that we may need another round of SDK/KC release which is time consuming.

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfc-gh-rcheng) [reply pom.xml]: using the fasterxml.version variable, thanks for the catch!

**Comment by sfc-gh-rcheng** [comment]
> Thanks Revi! Do you think we should provide a private jar to CITI or Informatica and ask them to run it with Blackduck in case there is still some differences? My concern is that we may need another round of SDK/KC release which is time consuming.

Agreed - discussed offline, lets ask informatica to run a blackduck scan for us. Created jira https://snowflakecomputing.atlassian.net/browse/SNOW-860497

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm, thanks!

**Comment by sfc-gh-rcheng** [comment]
Following error, so i think we need to retain the shaded hadoop jar. I'll add it back in, it was originally removed because I thought it caused the protobuf vulnerability. @sfc-gh-lsembera any thoughts?

`The valid symbol is in org.apache.hadoop.thirdparty:hadoop-shaded-protobuf_3_7:jar:1.1.1 at net.snowflake:snowflake-ingest-sdk:jar:2.0.2-SNAPSHOT / org.apache.hadoop:hadoop-common:3.3.5 (compile) / org.apache.hadoop.thirdparty:hadoop-shaded-protobuf_3_7:1.1.1 (compile) but it was not selected because net.snowflake:snowflake-ingest-sdk:2.0.2-SNAPSHOT excludes org.apache.hadoop.thirdparty:hadoop-shaded-protobuf_3_7.`

**Comment by sfc-gh-lsembera** [comment]
@sfc-gh-rcheng Putting versions into `<dependency>` does not have any effect because versions of these dependencies are already defined in `<dependencyManagement>` section and `dependencyManagement` is the source of truth for dependency versions. You can verify it by running `mvn dependency:list` with and without your change and see there are no differences (output is generated in `target/dependency-list.txt`.

The Hadoop errors seem to be caused by the linkage checker, you can fix it by adding a rule to `linkage-checker-exclusion-rules.xml`.

**Comment by sfc-gh-rcheng** [comment]
Thanks Lukas, youre right adding the version makes no change on the dependency list. Closing this PR as the versions are high enough to resolve the vulnerabilities

**Comment by sfc-gh-rcheng** [comment]
Discussed on slack - lets update all the packages in preparation for GA

**Pull request merged**